tests/db/migration_tests/test_v202310040_to_v202310090_.py

```diff
@@ -0,0 +1,34 @@
+"""
+    Copyright 2023 Inmanta
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Contact: code@inmanta.com
+"""
+import os
+from collections import abc
+
+import pytest
+
+from inmanta.data import Environment
+
+
+@pytest.mark.db_restore_dump(os.path.join(os.path.dirname(__file__), "dumps/v202310040.sql"))
+async def test_type_change(
+    migrate_db_from: abc.Callable[[], abc.Awaitable[None]], get_tables_in_db: abc.Callable[[], abc.Awaitable[list[str]]]
+) -> None:
+    await migrate_db_from()
```

Comment:
I would have liked to assert that the type is `int` before the migration but I didn't manage to

Reply:
I would expect you just need to same code as after the `migrate_from_db()` call but with `str` replaced by `int`?

Reply:
The old way of checking was raising an exception: calling this before the `migrate_from_db()` call would fail because of db pool exhaustion 
```py
env = await Environment.get_one(name="dev-1")
assert isinstance(env.settings["autostart_agent_deploy_interval"], int)
assert isinstance(env.settings["autostart_agent_repair_interval"], int)
```

Reply:
Do you know why we got the pool exhaustion. This is very strange.

Reply:
I find it strange as well, I have no idea